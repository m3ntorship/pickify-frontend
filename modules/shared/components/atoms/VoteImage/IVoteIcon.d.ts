declare namespace IVoteIcon {
  export interface IProps {
    isOneImage: boolean;
  }
}

export { IVoteIcon };
