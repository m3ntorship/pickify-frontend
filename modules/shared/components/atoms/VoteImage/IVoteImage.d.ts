declare namespace IVoteImage {
  export interface IProps {
    oneImage: boolean;
  }
}

export { IVoteImage };
