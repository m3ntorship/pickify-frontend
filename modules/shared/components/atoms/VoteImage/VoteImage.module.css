.love {
  @apply bg-primary hover:ring-4 transition-all ring-primary ring-opacity-25 w-10 h-10 focus:outline-none transform hover:scale-145 flex justify-center items-center rounded-full cursor-pointer;
}

.like {
  @apply bg-primary ml-4 hover:ml-8 hover:ring-4 transition-all ring-primary ring-opacity-25 w-10 h-10 focus:outline-none transform hover:scale-145 flex justify-center items-center rounded-full cursor-pointer;
}

.dislike {
  @apply bg-error hover:mr-4 hover:ring-4 transition-all ring-error ring-opacity-25 w-10 h-10 focus:outline-none transform hover:scale-145 flex justify-center items-center rounded-full cursor-pointer;
}
