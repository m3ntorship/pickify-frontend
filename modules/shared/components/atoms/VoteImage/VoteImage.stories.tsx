import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import VoteImage from './VoteImage';
import type { IVoteImage } from './IVoteImage';

export default {
  title: 'Atoms/VoteImage',
  component: VoteImage,
  argTypes: {
    oneImage: {
      control: {
        type: 'boolean',
      },
    },
  },
};

const Template: Story<IVoteImage.IProps> = (args): ReactElement => (
  <VoteImage {...args} />
);

export const VoteImageStory = Template.bind({});
VoteImageStory.args = {
  oneImage: false,
};
