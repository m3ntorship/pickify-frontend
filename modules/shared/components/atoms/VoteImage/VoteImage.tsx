import type { ReactElement, FC } from 'react';
import React from 'react';
import type { IVoteImage } from './IVoteImage';
import OutlinedHeartIcon from '../../icons/outlinedHeart.svg';
import FilledHeartIcon from '../../icons/filledHeart.svg';
import OutlinedDislikeIcon from '../../icons/outlinedDislike.svg';
import OutlinedLikeIcon from '../../icons/outlinedLike.svg';
import FilledLikeIcon from '../../icons/filledLike.svg';
import styles from './VoteImage.module.css';

const VoteImage: FC<IVoteImage.IProps> = ({ oneImage }): ReactElement => {
  return (
    <div className="relative">
      <div>
        {oneImage ? (
          <button type="button" className={`${styles.love} group`}>
            <OutlinedHeartIcon className="group-hover:hidden" />
            <FilledHeartIcon className="group-hover:block hidden" />
          </button>
        ) : (
          <div className="flex">
            <button type="button" className={`${styles.dislike} group`}>
              <OutlinedDislikeIcon />
            </button>
            <button type="button" className={`${styles.like} group`}>
              <OutlinedLikeIcon className="group-hover:hidden" />
              <FilledLikeIcon className="group-hover:block hidden" />
            </button>
          </div>
        )}
      </div>
    </div>
  );
};

export default VoteImage;
