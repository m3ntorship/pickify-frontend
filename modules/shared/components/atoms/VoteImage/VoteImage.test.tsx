import React from 'react';
import renderer from 'react-test-renderer';
import VoteImage from './VoteImage';

describe('snapshot testing', () => {
  it('should render heart button when oneImage is true', () => {
    const tree = renderer.create(<VoteImage oneImage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render two buttons when oneImage is true', () => {
    const tree = renderer.create(<VoteImage oneImage={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
