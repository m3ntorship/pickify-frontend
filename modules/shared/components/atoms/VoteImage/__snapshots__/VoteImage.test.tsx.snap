// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing should render heart button when oneImage is true 1`] = `
<div
  className="relative"
>
  <div>
    <button
      className="love group"
      type="button"
    >
      <svg
        className="group-hover:hidden"
        data-file-name="SvgOutlinedHeart"
      />
      <svg
        className="group-hover:block hidden"
        data-file-name="SvgFilledHeart"
      />
    </button>
  </div>
</div>
`;

exports[`snapshot testing should render two buttons when oneImage is true 1`] = `
<div
  className="relative"
>
  <div>
    <div
      className="flex"
    >
      <button
        className="dislike group"
        type="button"
      >
        <svg
          data-file-name="SvgOutlinedDislike"
        />
      </button>
      <button
        className="like group"
        type="button"
      >
        <svg
          className="group-hover:hidden"
          data-file-name="SvgOutlinedLike"
        />
        <svg
          className="group-hover:block hidden"
          data-file-name="SvgFilledLike"
        />
      </button>
    </div>
  </div>
</div>
`;
